palisade-service/src/main/java/uk/gov/gchq/palisade/service/palisade/service/ResourceService.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2019 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.service.palisade.service;
+
+import uk.gov.gchq.palisade.resource.LeafResource;
+import uk.gov.gchq.palisade.service.ConnectionDetail;
+import uk.gov.gchq.palisade.service.Service;
+import uk.gov.gchq.palisade.service.palisade.request.GetResourcesByIdRequest;
+import uk.gov.gchq.palisade.service.palisade.web.ResourceClient;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+
+public class ResourceService implements Service {
+
+    private final ResourceClient client;
+    private final Executor executor;
+
+    public ResourceService(final ResourceClient resourceClient, final Executor executor) {
+        this.client = resourceClient;
+        this.executor = executor;
+    }
+
+    public CompletionStage<Map<LeafResource, ConnectionDetail>> getResourcesById(final GetResourcesByIdRequest resource) {
```

Comment:
If we only have the getResourcesById method then we can delete the other resource request types

Reply:
@m09526 I am happy for us to bin off the other request types (by resource, by type, by serialised format). Are you happy with that change?

Reply:
How would someone request all files of one type then? If you can only request by ID, doesn't that mean we can only make a request for a known resource?

Reply:
It would mean that you need to know a directory path. I wonder whether it is wise to allow all resources of a type.